handler/src/test/java/io/netty/handler/ssl/SniHandlerTest.java

```diff
@@ -344,6 +344,42 @@ public void testFallbackToDefaultContext() throws Exception {
         }
     }
 
+    @Test(timeout = 10000)
+    public void testMajorVersionNot3() throws Exception {
+        SslContext nettyContext = makeSslContext(provider, false);
+
+        try {
+            DomainNameMapping<SslContext> mapping = new DomainNameMappingBuilder<SslContext>(nettyContext).build();
+
+            SniHandler handler = new SniHandler(mapping);
+            EmbeddedChannel ch = new EmbeddedChannel(handler);
+
+            // invalid
+            byte[] message = {22, 2, 0, 0, 0};
+            try {
+                // Push the handshake message.
+                ch.writeInbound(Unpooled.wrappedBuffer(message));
+            } catch (Exception e) {
```

Comment:
add `fail();` above

Reply:
@normanmaurer done, I actually just copied the existing `testFallbackToDefaultContext()` as a starting point, which had this `fail()` commented out for some reason. So I've uncommented it there too since an exception does also get thrown in that case.

Reply:
@njhill after some more digging this actually depends on if `jdkCompatibilityMode` is true of not. Let me do a follow up to fix it and then investigate later on.